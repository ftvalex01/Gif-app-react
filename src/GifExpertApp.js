import React from 'react'
import { useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories)

    

  return (
      <>
      <div className='container'>
      <h2 className='header'>GifExpertApp</h2>
      <AddCategory  setCategories={setCategories}></AddCategory>
      <hr/>
      </div>
      <ol>
          {categories.map(category =>{
              return <GifGrid key={category} category={category}/>
          })}
      </ol>
      </>
  )
}

export default GifExpertApp