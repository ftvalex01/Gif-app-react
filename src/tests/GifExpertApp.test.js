import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en el GifExpertApp', () => {
    
    test('Haciendo match con el snapshot', () => {
        
        const wrapper = shallow(<GifExpertApp/>)
        expect(wrapper).toMatchSnapshot()


    });

    test('debe de mostrar una lista de categorias', () => {
        
        const categories = ['One Punch','Dragon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    }); 

});