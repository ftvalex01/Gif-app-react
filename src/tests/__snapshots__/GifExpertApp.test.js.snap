// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifExpertApp Haciendo match con el snapshot 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h2
      className="header"
    >
      GifExpertApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
  </div>
  <ol />
</Fragment>
`;

exports[`Pruebas en el GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h2
      className="header"
    >
      GifExpertApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
  </div>
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;
